mlrun/utils/notifications/notification_pusher.py

```diff
@@ -32,7 +32,55 @@
 from .notification import NotificationBase, NotificationTypes
 
 
-class NotificationPusher(object):
+class _NotificationPusherBase(object):
+    def _push(
+        self, sync_push_callback: typing.Callable, async_push_callback: typing.Callable
+    ):
+
+        if mlrun.utils.helpers.is_running_in_jupyter_notebook():
+            # Running in Jupyter notebook.
+            # In this case, we need to create a new thread, run a separate event loop in
+            # that thread, and use it instead of the main_event_loop.
+            # This is necessary because Jupyter Notebook has its own event loop,
+            # but it runs in the main thread. As long as a cell is running,
+            # the event loop will not execute properly
+            self._run_coroutine_in_jupyter_notebook(
+                coroutine_method=async_push_callback
+            )
+        else:
+            # Either running in mlrun api or sdk. in case of mlrun api we are in a separate thread from the one in which
+            # the main event loop. we cannot use main loop, so creating a new one.
+            event_loop = asyncio.new_event_loop()
```

Comment:
why not doing a "get or create" event loop to ensure we are not creating too many